apps/studio-next/src/app/api/crawler/route.tsx

```diff
@@ -0,0 +1,100 @@
+import { NextRequest, NextResponse } from "next/server";
+import parseURL from "@/helpers/parser";
+import { DocumentInfo } from "@/types";
+import axios from "axios";
+
+export async function GET(request: NextRequest) {
+  const Base64searchParams = request.nextUrl.searchParams.get('base64');
+  const URLsearchParams = request.nextUrl.searchParams.get('url');
+
+  try {
+    if (!Base64searchParams && !URLsearchParams) return new NextResponse(null, { status: 200 });
+    let info: DocumentInfo | null = null;
+
+    if (Base64searchParams) {
+      // directly run the parsing function
+      info = await parseURL(Base64searchParams);
+    }
+    if (URLsearchParams) {
+      // fetch the document information from the URL
+      try {
+          const response = await axios.get(URLsearchParams);
+          if (response.status === 200) {
+            info = await parseURL(response.data);
+          } else {
+            return new NextResponse("Not a valid URL", { status: 500 });
+          }
+      } catch (error) {
+        return new NextResponse("Not a valid URL", { status: 500 });
+      }
+    }
+
+    if (!info) {
+      const crawlerInfo = `
+       <!DOCTYPE html>
+       <html lang="en">
+       <head>
+        <meta charset="UTF-8">
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <title>"AsyncAPI Studio"</title>
+        <meta property="og:title" content="AsyncAPI Studio" />
+        <meta property="og:description" content="Studio for AsyncAPI specification, where you can validate, view preview documentation, and generate templates from AsyncAPI document." />
+        <meta property="og:url" content="https://studio-next.netlify.app" />
+        <meta property="og:image" content="https://raw.githubusercontent.com/asyncapi/studio/master/apps/studio-next/public/img/meta-studio-og-image.jpeg" />
```

Comment:
Can those values be grabbed from https://github.com/asyncapi/studio/blob/a428b15f6993687012396961e0032d14ee17ebee/apps/studio-next/src/app/page.tsx#L12 ?

Reply:
Done, @smoya